### PR TITLE
[mailstream] various modernisations

### PR DESCRIPTION
**Pull request by mexon**
* Remove unused "cron" hook
* Add "App" arguments to hooks where that is one of the arguments
* Replace "getHost" with "getHostname"
* Allow items with an empty plink
* Remove "ANNOUNCE" items
* Use baseUrl()->get() instead of baseUrl()
* Remove in-place modification in mailstream_html_wrap
* Tweaks to logging

### CODE REVIEW

**Comment by mexon** [comment]
Could someone remind me how to update translation files?  I added one translatable string, "URI".  The translation, into every human language known, should read "URI" 😉

**Comment by MrPetovan** [comment]
Please run `bin/run_xgettext.sh -a mailstream` from the base Friendica folder.

**Comment by mexon** [comment]
> Please run `bin/run_xgettext.sh -a mailstream` from the base Friendica folder.

Done, thanks for the reminder!

File: mailstream/lang/C/messages.po [hunk 1]
```diff
@@ -1,20 +1,20 @@
-# ADDON mailstream
-# Copyright (C) 
-# This file is distributed under the same license as the Friendica mailstream addon package.
-# 
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
```
Comment (MrPetovan) [inline mailstream/lang/C/messages.po:4]: This doesn't look right, please revert to the following version:
```suggestion
# ADDON mailstream
# Copyright (C) 2023 The Friendica Project
# This file is distributed under the same license as the Friendica mailstream addon package.
#
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mailstream/lang/C/messages.po (hunk 1 above)
Comment (MrPetovan) [inline mailstream/lang/C/messages.po]: This doesn't look right, please revert the changes to these lines.

File: mailstream/lang/C/messages.po [hunk 2]
```diff
@@ -1,20 +1,20 @@
...
 #
 #, fuzzy
 msgid ""
 msgstr ""
-"Project-Id-Version: \n"
+"Project-Id-Version: PACKAGE VERSION\n"
```
Comment (MrPetovan) [inline mailstream/lang/C/messages.po]: Same here.

File: mailstream/lang/C/messages.po [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2021-11-21 19:15-0500\n"
+"POT-Creation-Date: 2023-04-01 15:07+0200\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
 "Language: \n"
 "MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Type: text/plain; charset=CHARSET\n"
```
Comment (MrPetovan) [inline mailstream/lang/C/messages.po]: Same here.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: mailstream/lang/C/messages.po (hunk 1 above)
Comment (mexon) [reply mailstream/lang/C/messages.po]: Done.

File: mailstream/lang/C/messages.po (hunk 2 above)
Comment (mexon) [reply mailstream/lang/C/messages.po]: Done

File: mailstream/lang/C/messages.po (hunk 3 above)
Comment (mexon) [reply mailstream/lang/C/messages.po]: Done

File: mailstream/mailstream.php [hunk 4]
```diff
@@ -138,15 +138,16 @@ function mailstream_send_hook(array $data)
...
  * @return void
  */
-function mailstream_post_hook(array &$item)
+function mailstream_post_hook(App $a, array &$item)
 {
 	mailstream_check_version();
 
 	if (!DI::pConfig()->get($item['uid'], 'mailstream', 'enabled')) {
-		Logger::debug('mailstream: not enabled for item ' . $item['id']);
+		Logger::debug('mailstream: not enabled for item ' . $item['id'] . ' uid ' . $item['uid']);
```
Comment (nupplaphil) [inline mailstream/mailstream.php]:
```suggestion
		Logger::debug('mailstream: not enabled.', ['item' => $item['id'], ' uid ' => $item['uid']]);
```

File: mailstream/mailstream.php [hunk 5]
```diff
@@ -161,8 +162,8 @@ function mailstream_post_hook(array &$item)
 		Logger::debug('mailstream: no uri for item ' . $item['id']);
 		return;
 	}
-	if (!$item['plink']) {
-		Logger::debug('mailstream: no plink for item ' . $item['id']);
+	if ($item['verb'] == Activity::ANNOUNCE) {
+		Logger::debug('mailstream: announce item ' . $item['id']);
```
Comment (nupplaphil) [inline mailstream/mailstream.php]:
```suggestion
		Logger::debug('mailstream: announce item ', ['item' => $item['id']]);
```

File: mailstream/mailstream.php [hunk 6]
```diff
@@ -462,7 +468,7 @@ function mailstream_convert_table_entries()
 					'message_id' => $ms_item_id['message-id'],
 					'tries' => 0);
 		if (!$ms_item_id['message-id'] || !strlen($ms_item_id['message-id'])) {
-			Logger::info('mailstream_cron: Item ' .
+			Logger::info('mailstream_convert_table_entries: Item ' .
 							$ms_item_id['id'] . ' URI ' . $ms_item_id['uri'] . ' has no message-id');
```
Comment (nupplaphil) [inline mailstream/mailstream.php]:
```suggestion
			Logger::info('mailstream_convert_table_entries: item has no message-id.', 'item' => $ms_item_id['id'], 'uri' => $ms_item_id['uri']]);
```

**Review by nupplaphil — CHANGES_REQUESTED** [review]
some small logging improvements :)

**Review by MrPetovan — APPROVED** [review]
